pkg/config/kpconfig.go

```diff
@@ -99,57 +134,53 @@ func (d KpConfigProvider) SetDefaultServiceAccount(ctx context.Context, serviceA
 	}
 
 	if k8serrors.IsNotFound(err) {
-		return createKpConfigMap(ctx, d.cs.K8sClient, KpConfig{
+		return d.createKpConfigMap(ctx, KpConfig{
 			serviceAccount: serviceAccount,
 		})
 	}
 
-	return updateKpConfigMap(ctx, d.cs.K8sClient, existingKpConfig, KpConfig{serviceAccount: serviceAccount})
+	return d.updateKpConfigMap(ctx, existingKpConfig, KpConfig{serviceAccount: serviceAccount})
 }
 
 func (d KpConfigProvider) getKpConfigMap(ctx context.Context) (*corev1.ConfigMap, error) {
-	return d.cs.K8sClient.CoreV1().ConfigMaps(kpNamespace).Get(ctx, kpConfigMapName, metav1.GetOptions{})
+	return d.client.CoreV1().ConfigMaps(kpConfigNamespace).Get(ctx, kpConfigMapName, metav1.GetOptions{})
 }
 
-func configMapFromKpConfig(config KpConfig) *corev1.ConfigMap {
+func (d KpConfigProvider) configMapFromKpConfig(config KpConfig) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      kpConfigMapName,
-			Namespace: kpNamespace,
+			Namespace: kpConfigNamespace,
 		},
 		Data: map[string]string{
-			defaultRepositoryKey:              config.defaultRepository,
-			defaultServiceAccountNameKey:      config.serviceAccount.Name,
-			defaultServiceAccountNamespaceKey: config.serviceAccount.Namespace,
+			d.repoKey:        config.defaultRepository,
+			d.saNameKey:      config.serviceAccount.Name,
+			d.saNamespaceKey: config.serviceAccount.Namespace,
 		},
 	}
 }
 
-func createKpConfigMap(ctx context.Context, client kubernetes.Interface, config KpConfig) error {
-	_, err := client.CoreV1().ConfigMaps(kpNamespace).Create(ctx, configMapFromKpConfig(config), metav1.CreateOptions{})
-	if err != nil {
-		return err
-	}
-
-	return nil
+func (d KpConfigProvider) createKpConfigMap(ctx context.Context, config KpConfig) error {
+	_, err := d.client.CoreV1().ConfigMaps(kpConfigNamespace).Create(ctx, d.configMapFromKpConfig(config), metav1.CreateOptions{})
+	return err
 }
 
-func updateKpConfigMap(ctx context.Context, client kubernetes.Interface, existingConfig *corev1.ConfigMap, newConfig KpConfig) error {
+func (d KpConfigProvider) updateKpConfigMap(ctx context.Context, existingConfig *corev1.ConfigMap, newConfig KpConfig) error {
 	updatedConfig := existingConfig.DeepCopy()
 
 	if newConfig.defaultRepository != "" {
-		updatedConfig.Data[defaultRepositoryKey] = newConfig.defaultRepository
+		updatedConfig.Data[d.repoKey] = newConfig.defaultRepository
```

Comment:
Shouldn't we write the `default.repository` and `canonical.repository"` always? This provides backwards compatibility if old versions of `kp` operate on the cluster and positions ourselves to eventually deprecate those keys in a future version of `kp`.